src/irmin-pack/unix/dispatcher_intf.ml

```diff
@@ -80,6 +80,10 @@ module type S = sig
   val read_in_prefix_and_suffix_exn : t -> off:int63 -> len:int -> bytes -> unit
   (** Simlar to [read_exn] but if [off + len] is greater than the end of the
       prefix, it will read the remaining in the prefix. *)
+
+  val poff_of_entry_exn : Mapping_file.t -> off:int63 -> len:int -> int63
+  (** [poff_of_entry_exn] converts a global offset into a physical prefix
+      offset. *)
```

Comment:
The name of this function doesn't convey very well what it does and it is also pretty redundant with `create_accessor_exn`.

I suggest you use `create_accessor_exn` and privately reveal the accessor type:
```diff
-  type accessor
+  type location = private Prefix | Suffix [@@deriving irmin]
+  type accessor = private { poff : int63; len : int; location : location } [@@deriving irmin]
```

Reply:
I don't think `create_accessor_exn` works -  the mapping used in the dispatcher is the old mapping, whereas here we need to use the new one, before it being set as the default mapping in the file manager.

Reply:
True. 

In that case I suggest a `val create_accessor_to_prefix_exn : Mapping_file.t -> off:int63 -> len:int -> accessor`.

Reply:
done, thanks